system-addon/lib/PlacesFeed.jsm

```diff
@@ -262,6 +264,39 @@ class PlacesFeed {
     win.openLinkIn(action.data.url, where || win.whereToOpenLink(event), params);
   }
 
+  addLinkToPocket(url, title) {
+    return new Promise((resolve, reject) => {
+      pktApi.addLink(url, {
+        title,
+        success(data) {
+          resolve(data.item);
+        },
+        error(err) {
+          reject(err);
+        }
+      });
+    });
+  }
+
+  async saveToPocket(action) {
+    const {url, title} = action.data.site;
+    // If the user is not logged in, show the panel to prompt them to log in
+    if (!pktApi.isUserLoggedIn()) {
+      Pocket.savePage(action._target.browser, url, title);
+      return;
+    }
+    try {
+      // If the user is logged in, just save the link to Pocket and update the page
+      let {item_id} = await this.addLinkToPocket(url, title);
```

Comment:
What is the expected behavior of saving to pocket? Before, it would open up the tagging doorhanger and trigger the library animation.

Reply:
Note that the bookmark behavior is to open the bookmark doorhanger that triggers the library animation.

Reply:
While we wait for the ui wanted for the correct behaviour, I'll explain my reasoning behind this change..
calling ```Pocket.savePage``` although does drop the doorhanger down doesn't have any indication of a successful or not successful save once it goes off and does it's stuff. With the pktApi we get the pocket_id back when we're successful so we can update the card we just saved, similar to how we do that for bookmark (i.e changing the context menu + highlight context) So I traded off one kind of confirmation feedback for a different one. Ideally we would call ```addLinkToPocket```, and then also call something in Pocket.jsm to *just* show the doorhanger + library animation.

Reply:
The designers say don't bother with the doorhanger :D